analyzers/maven/maven.go

```diff
@@ -73,19 +75,46 @@ func (a *Analyzer) Discover(dir string) ([]module.Module, error) {
 
 			dir := filepath.Dir(path)
 			submodules, err := a.Maven.Modules(dir)
-			if err != nil {
-				log.Logger.Debugf("Modules err: %#v %#v", err.Error(), err)
-				return err
-			}
+			if len(submodules) == 0 {
```

Comment:
We should also explicitly check whether `err != nil`. If `err != nil`, then `submodules` is not guaranteed to be 0, and using its values is UB.